uxarray/plot/accessor.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Any, overload, Optional
+
+import functools
+
+if TYPE_CHECKING:
+    from uxarray.core.dataset import UxDataset
+    from uxarray.core.dataarray import UxDataArray
+    from uxarray.grid import Grid
+
+import uxarray.plot.grid_plot as grid_plot
+import uxarray.plot.dataarray_plot as dataarray_plot
+
+
+class GridPlotAccessor:
+    _uxgrid: Grid
+    __slots__ = ("_uxgrid",)
+
+    def __init__(self, uxgrid: Grid) -> None:
+        self._uxgrid = uxgrid
+
+    @functools.wraps(grid_plot.plot)
+    def __call__(self, **kwargs) -> Any:
+        return grid_plot.plot(self._uxgrid, **kwargs)
+
+    @functools.wraps(grid_plot.nodes)
+    def nodes(self, **kwargs):
+        return grid_plot.nodes(self._uxgrid, **kwargs)
+
+    @functools.wraps(grid_plot.edges)
+    def edges(self, **kwargs):
+        return grid_plot.edges(self._uxgrid, **kwargs)
+
+
+class UxDataArrayPlotAccessor:
+    _uxda: UxDataArray
+    __slots__ = ("_uxda",)
+
+    def __init__(self, uxda: UxDataArray) -> None:
+        self._uxda = uxda
+
+    @functools.wraps(dataarray_plot.plot)
+    def __call__(self, **kwargs) -> Any:
+        return dataarray_plot.plot(self._uxda, **kwargs)
+
+    @functools.wraps(dataarray_plot.raster)
+    def raster(self,
+               plot_height: Optional[int] = 300,
+               plot_width: Optional[int] = 600,
+               cmap: Optional[str] = "blue",
+               agg: Optional[str] = "mean"):
+        """TODO: Docstring & additional params"""
+        return dataarray_plot.raster(self._uxda, plot_height, plot_width, cmap,
+                                     agg)
```

Comment:
@erogluorhan 

For the scope of this PR, I've set up the accessor and modules for interfacing with our plotting methods. Additionally, I've also set up a `UxDataArray.raster()` method directly using Datashader to perform the visualization. 

If you are good with moving forward with this PR only including the `UxDataArray.raster()` method, we can start implementing all other methods. I was planning on beginning with our `Grid` plotting methods, however after our discussion about the issues we've encountered, we need to spend some time figuring those out. 

Let me know what you think, planning on getting this PR polished and ready for review today.